xbmc/utils/StringUtils.h

```diff
@@ -92,6 +93,19 @@ class StringUtils
   static bool EndsWithNoCase(const std::string &str1, const char *s2);
 
   static std::string Join(const std::vector<std::string> &strings, const std::string& delimiter);
+  template <typename T, typename U>
+  static std::string Join(const std::map<T, U>& input_map, const std::string& delimiter = ",") {
+    if (input_map.empty())
+      return "";
+
+    std::ostringstream ss;
+    for (auto const& element : input_map)
+      ss << element.first << delimiter;
```

Comment:
I tested for speed and to_string is way faster but I already find out in another pr that doesn't compile in android..
I guess that's the reason we use `StringUtils::Format` everywhere. 
So I tested format and stringstream and stringstream was still faster with big maps.
Do you know why it doesn't work in android?